libs/sdk-core/src/models.rs

```diff
@@ -982,6 +987,52 @@ pub struct SendOnchainResponse {
     pub reverse_swap_info: ReverseSwapInfo,
 }
 
+pub enum DesiredSwapAmountType {
+    Send,
+    Receive,
+}
+
+/// See [ReverseSwapFeesRequest]
+pub struct PrepareOnchainPaymentRequest {
+    /// Depending on `amount_type`, this may be the desired send amount or the desired receive amount.
+    pub amount_sat: u64,
+    pub amount_type: DesiredSwapAmountType,
+
+    /// Feerate (sat / vByte) for the claim transaction
+    pub claim_tx_feerate: u32,
+}
+
+/// Contains fields describing the reverse swap parameters (see [ReverseSwapPairInfo]), as well as
+/// the resulting send and receive amounts.
+///
+/// The `send_amount_sat` and `receive_amount_sat` are empty if the `send_amount_sat` falls outside
+/// the min / max range.
+#[derive(Debug)]
+pub struct PrepareOnchainPaymentResponse {
+    // Fields from ReverseSwapPairInfo:
+    pub min: u64,
+    pub max: u64,
+    pub fees_hash: String,
+    pub fees_percentage: f64,
+    pub fees_lockup: u64,
+    pub fees_claim: u64,
+
+    // Fields set only if the resulting send amount is within range
+    pub send_amount_sat: Option<u64>,
+    pub receive_amount_sat: Option<u64>,
+}
+
+pub struct PayOnchainRequest {
+    pub send_amount_sat: u64,
+    pub receive_amount_sat: u64,
```

Comment:
Shouldn't these be optional?

Reply:
No because they are both returned in `PrepareOnchainPaymentResponse`.

Reply:
It's a bit confusing on the app side, let's say I'd like to prepare onchain payment for sending 50_000 sats, why would I also pass a value to receive amount sat?

Reply:
The order would be
- `prepare_onchain_payment` with `amount_sat = 50 000` and `amount_type = Send`
- this returns the intermediary fees, `send_amount_sat` and `receive_amount_sat`
- the app can then call `pay_onchain` with the send and receive amounts

Does this address your question?

Reply:
@ok300 we had a discussion today and Erdem has a point. The relation between the prepareOnchainResponse and the payOnchain input is not concise and has some edge cases that are not so clear to a new user.
The suggestion was:
1. The prepareOnchain will return PayOnchainRequest
2. Then the user should call payOnchain with the above request so it is clear how to use the whole flow.

@erdemyerebasmaz this is what you had in mind?